### DISCUSSION TITLE
Change custom headers in a part of a NextJS App

### DISCUSSION THREAD

**Discussion by psteinroe** [supabase/supabase#20048] [Questions]
Hey, we are using supabase auth helpers in a NextJs App using the Pages Router. Our setup is exactly the one described in the docs. Most importantly, we create a browser client in `_app`:

```tsx
  const [supabase] = useState(() =>
    createPagesBrowserClient<Database>({
      options: { global: { headers: { 'x-mateo-app': 'web' } } },
    })
  );
```

As you can see, we add a custom header to declare the app. We now want to change the header for a part of the app. Specifically, we want the header to have a different value on a specific path. 

Is there a way to either
- change the header for a client reference or
- set the header for a single rpc call?

There won't be any api for this, but maybe someone knows a workaround?

Best
Philipp

**Comment by GaryAustin1** [comment]
I know with standard supabase-js you can just create extra browser clients and use one with a header and the other not.
I don't know if that can be done with Auth-helpers.

At one time before there was a .schema() option, I did this to set a header for each call.  Once again on supabase-js.
```js
let schema = ''
SUPABASE_KEY = 
   let supabase1 = supabase.createClient(SUPABASE_URL, SUPABASE_KEY, {
    global: {
        fetch: (requestInfo, requestInit) => {
            if (schema) requestInit.headers['Accept-Profile'] = schema
            return(fetch(requestInfo,requestInit))
        },
        ....
 ```
